module/solutions/helper.js

```diff
@@ -462,7 +465,7 @@ module.exports = class SolutionsHelper {
    * @returns {JSON} - Auto targeted solutions query.
    */
 
-  static queryBasedOnRoleAndLocation(data, type = '') {
+  static queryBasedOnRoleAndLocation(data, type = '',origin,userToken) {
```

Comment:
@MallanagoudaB handle if origin is not passed. better to have validation for header origin